docs: update enterprise licensing description (#13871)
hashicorp/consul: website/content/docs/enterprise/license/overview.mdx

agents could start without a license and then have a license applied to them later on via the CLI
or API. That functionality has been removed and replaced with the ability to load licenses from the
agent's configuration or environment. Also, prior to 1.10.0, server agents would automatically propagate
the license between themselves. This no longer occurs and the license must be present on each server agent
when it is started.

-> Visit the [Enterprise License Tutorial](https://learn.hashicorp.com/tutorials/nomad/hashicorp-enterprise-license?in=consul/enterprise) for detailed steps on how to install the license key.

### Applying a License

For Consul Enterprise 1.10.0 or greater, a license must be available at the time the agent starts.
For server agents this means that they must either have the [`license_path`](/docs/agent/config/config-files#license_path)
configuration set or have a license configured in the servers environment with the `CONSUL_LICENSE` or
`CONSUL_LICENSE_PATH` environment variables. Both the configuration item and the `CONSUL_LICENSE_PATH`
environment variable point to a file containing the license whereas the `CONSUL_LICENSE` environment

> variable should contain the license as the value. If multiple of these are set, the order of precedence is:

Reviewer: ```suggestion
variable should contain the license as the value. If multiple variables are set, the following order of precedence applies:
```

1. `CONSUL_LICENSE` environment variable
2. `CONSUL_LICENSE_PATH` environment variable
3. `license_path` configuration item.

> Both client agents and the snapshot agent may also be licensed in the very same manner. However, to prevent

Reviewer: ```suggestion
Client and snapshot agents may also be licensed in the same manner. However, to prevent
```
